docs/whirlpool/docs/03-Whirlpools SDKs/01-Whirlpools/06-Send Transaction.mdx

```diff
@@ -0,0 +1,372 @@
+    ```
+  </TabItem>
+  <TabItem value="rust" label="Rust">
+    ```rust title="main.rs"
+    #[tokio::main]
+    async fn main() {
+      // ...
+      let whirlpool_address =
+      Pubkey::from_str("Hxw77h9fEx598afiiZunwHaX3vYu9UskDk9EpPNZp1mG").unwrap();
+
+      let increase_liquidity_param = IncreaseLiquidityParam::TokenB(100_000);
+      let open_position_instructions = open_position_instructions(
+        &rpc,
+        whirlpool_address,
+        30.0,
+        70.0,
+        increase_liquidity_param,
+        Some(100),
+        Some(wallet.pubkey()),
+      )
+      .await
+      .unwrap();
+      // ...
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+### 4: Create Transaction Message
+To send a transaction on Solana, you need to include a blockhash. A blockhash acts as a timestamp and ensures the transaction has a limited lifetime. Validators use the blockhash to verify the recency of a transaction before including it in a block. A transaction referencing a blockhash is only valid for 150 slots (~1-2 minutes, depending on slot time). After that, the blockhash expires, and the transaction will be rejected.
+
+Here's how the transaction message is created:
+
+<Tabs groupId="programming-languages">
+  <TabItem value="ts" label="Typescript" default>
+    ```tsx title="sendTransaction.ts"
+    const latestBlockHash = await rpc.getLatestBlockhash().send();
+    const transactionMessage = await pipe(
+      createTransactionMessage({ version: 0}),
+      tx => setTransactionMessageFeePayer(wallet.address, tx),
+      tx => setTransactionMessageLifetimeUsingBlockhash(latestBlockHash.value, tx),
+      tx => appendTransactionMessageInstructions(instructions, tx)
+    )
+    ```
+  </TabItem>
+  <TabItem value="rust" label="Rust">
+    ```rust title="main.rs"
+    #[tokio::main]
+    async fn main() {
+      // ...
+      let message = Message::new(
+        &open_position_instructions.instructions,
+        Some(&wallet.pubkey()),
+      );
+      let mut signers: Vec<&dyn Signer> = vec![&wallet];
+      signers.extend(
+        open_position_instructions
+          .additional_signers
+          .iter()
+          .map(|kp| kp as &dyn Signer),
+      );
+      let recent_blockhash = rpc.get_latest_blockhash().await.unwrap();
+      let transaction = Transaction::new(&signers, message, recent_blockhash);
+      // ...
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+### 5: Estimating Compute Unit Limit and Prioritization Fee
+When preparing a transaction, it's important to set a compute unit limit and an appropriate prioritization fee. Setting the compute units too low and the transaction will fail. Setting it too high will make the transaction less favorable for validators to process..
+
+You can get an estimation of the compute units by simulating the transaction on the RPC. To avoid transaction failures caused by underestimating this limit, an additional 100,000 compute units are added, but you can adjust this based on your own tests.
```

Comment:
Might be good to add here that txs that request fewer compute units get high priority for the same amount of prio fee. Thus lower = better but too low and the tx fails